datafusion/functions-array/src/string.rs

```diff
@@ -281,6 +281,24 @@ pub(super) fn array_to_string_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
 
                 Ok(arg)
             }
+            Dictionary(..) => {
+                let any_dict_array = arr
+                    .as_any_dictionary_opt()
+                    .ok_or_else( || {
+                        DataFusionError::Internal(
+                            format!("could not cast {} to AnyDictionaryArray", arr.data_type())
+                        )
+                    })?;
+                compute_array_to_string(
```

Comment:
I pushed some tests showing how this results in incorrect values.

I think you may be able to call `to_string!` directly to iterate over the values of the dictionary.

Reply:
thanks for the feedback, let me figure it out.

Reply:
@alamb thanks for the tests, I think I have a better understanding of DictionaryArray. 

Based on my understanding, I converted dictionary keys and values to something iterable. Then using the keys to access the values I got the logical array. I think tests are now showing correct results.

Finally, I am happy to work on any new feedback I receive.

Reply:
Amazing @EduardoVega  -- thank you so much -- this is quite an impressive first contribution. I didn't realize it was going to be quite as tricky.